smartapps/samsung/sleepsense-thermostat-control.src/sleepsense-thermostat-control.groovy

```diff
@@ -0,0 +1,173 @@
+	// TODO: subscribe to attributes, devices, locations, etc.
+}
+
+
+def sleepsensorHandler(evt) {
+	//log.debug "sleepsensorHandler: $evt.name"	
+    log.debug "sleepsensorHandler: $evt.value"
+    state.isOK = false;
+    
+    if(evt.value == "in bed")
+    {
+    	log.debug "In Bed actions will be excuted"
+        state.sleepAction = "in bed"
+    	state.isOK = true;
+    }
+    else if(evt.value == "out of bed")
+    {
+    	log.debug "Out Bed actions will be excuted"
+        state.sleepAction = "Out Bed"
+    	state.isOK = true;
+    }
+    else if(evt.value == "sleeping")
+    {
+    	log.debug "Sleeping actions will be excuted"
+        state.sleepAction = "Sleeping"
+    	state.isOK = true;
+    }
+    else if(evt.value == "not sleeping")
+    {
+    	log.debug "Awake actions will be excuted"
+      	state.sleepAction = "Awake"
+    	state.isOK = true;
+    }
+    else
+    {
+    	return
+    }
+    
+    thermostatcontrol()
+   
+}
+
+
+
+def thermostatcontrol() {
+    if(state.isOK == true)
+    {    	
+       if(state.sleepAction == "in bed")
+        {
+            log.debug "In Bed temperature control"
+            thermostatcontrol.each
+            {
+            	log.trace inbed_temperature
+                log.trace thermostatcontrol                
+	            def value = inbed_temperature as Double
+                thermostatcontrol.setCoolingSetpoint(value)
+                
+            }            
+        }
+        else if(state.sleepAction == "Out Bed")
+        {
+            log.debug "Out Bed temperature control"
+            thermostatcontrol.each
+            {
+	            log.trace inbed_temperature
+    	        log.trace thermostatcontrol               
+                if(outofbed_temperature == null)
+                {
+	    	        log.trace "input data null"
+                }
+                else
+                {
+                    def value = outofbed_temperature as Double
+                    thermostatcontrol.setCoolingSetpoint(value)
+                }
+            }
+        }
+        else if(state.sleepAction == "Sleeping")
+        {
+            log.debug "Sleeping temperature control"
+            thermostatcontrol.each
+            {
+	            log.trace inbed_temperature
+    	        log.trace thermostatcontrol                
+        	    def value = sleep_temperature as Double
+            	thermostatcontrol.setCoolingSetpoint(value)
+            }
+        }
+        else if(state.sleepAction == "Awake")
+        {
+            log.debug "Awake temperature control"
+            thermostatcontrol.each
+            {
+	            log.trace inbed_temperature
+    	        log.trace thermostatcontrol                
+        	    def value = wake_temperature as Double
+            	thermostatcontrol.setCoolingSetpoint(value)
+            }
+        }
+    }
+}
+
+def changedLocationMode(evt) {
+	log.debug "changedLocationMode: $evt"
+	switches?.on()
+}
+
+def appTouch(evt) {
```

Comment:
This isn't the proper use of this. See comments in previous submitted files above.